pkgs/by-name/te/telegram-macos/package.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenvNoCC,
+  unzip,
+  fetchurl,
+}:
+let
+  version = "10.15.2-265540";
+  versionTokens = lib.split "-" version;
+  versionFirst = lib.head versionTokens;
+  versionSecond = lib.last versionTokens;
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit version;
+
+  pname = "telegram-macos";
+
+  src = fetchurl {
+    url = "https://osx.telegram.org/updates/Telegram-${versionFirst}.${versionSecond}.app.zip";
+    hash = "sha256-ne087xxFihsv+3UMFFk3zGzaweRG9o8mVwXkMIwH/9Y=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/Applications
+    cp -r Telegram.app $out/Applications
+    runHook postInstall
+  '';
+
+  meta = {
```

Comment:
Please add `hydraPlatforms = []`, this prevents Hydra from caching the unpacked application bundle.

Reply:
@toonn This change was not well received in a few of my previous PRs, i.e. here's an example

https://github.com/NixOS/nixpkgs/pull/324145#discussion_r1667402816

It would be good to clarify what is the best approach here 🙏🏻

Reply:
cc @SuperSandro2000, any input on this will be greatly appreciated

Reply:
If I am not wrong, it should be fine to cache it if it's redistributable, packages with `lib.licenses.unfree` are not cached at all.

Reply:
I'm not sure why @SuperSandro2000 doesn't like setting hydraPlatforms empty, an explanation might be enligtening.

It's not about the license being redistributable or not. As you point out, unfree licensed software is already not built and cached by Hydra. It's about not wasting Hydra cycles on fetching and unpacking an upstream archive to then repack it into a NAR and take up binary cache space basically forever under the current retention policy. Cache sponsorship is contingent on keeping growth under control and this is an incredibly easy way to do so.